### PR TITLE
[plugin.video.embycon] 1.4.39

### PR DESCRIPTION
**Pull request by faush01**
### Description

Latest release of EmbyCon (1.4)

 - Lots of code refactoring
 - Removed some redundant code
 - Change the way playback works, now uses a Notification service to listen for play events
 - Send Notifications for Play/Display/Stop actions
 - Add error reporting prompt
 - Refactor the way logging works
 - Add music support

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.embycon/resources/lib/websocket.py [hunk 1]
```diff
@@ -0,0 +1,930 @@
...
+        else:
+            mask_key = self.get_mask_key(4)
+            return frame_header + self._get_masked(mask_key)
+
+    def _get_masked(self, mask_key):
+        s = ABNF.mask(mask_key, self.data)
+        return mask_key + "".join(s)
+
+    @staticmethod
+    def mask(mask_key, data):
```
[the remark this answers is not in the document]
Comment (faush01) [reply plugin.video.embycon/resources/lib/websocket.py]: 3rd party code, not going to change

**Comment by Lunatixz** [comment]
LGTM

File: plugin.video.embycon/resources/lib/websocket.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (faush01) [reply plugin.video.embycon/resources/lib/websocket.py]: this is 3rd party code which I would prefer not to touch

[the remark this answers is not in the document]
Comment (faush01) [reply plugin.video.embycon/resources/lib/websocket.py]: I fixed this one

**Comment by faush01** [comment]
any chance we can dig someone up to do a second review

**Pull request merged**